oauth.go

```diff
@@ -0,0 +1,149 @@
+	BusinessType       OAuthStripeUserBusinessType `form:"business_type"`
+	City               *string                     `form:"city"`
+	Country            *string                     `form:"country"`
+	Currency           *string                     `form:"currency"`
+	DobDay             uint64                      `form:"dob_day"`
+	DobMonth           uint64                      `form:"dob_month"`
+	DobYear            uint64                      `form:"dob_year"`
+	Email              *string                     `form:"email"`
+	FirstName          *string                     `form:"first_name"`
+	FirstNameKana      *string                     `form:"first_name_kana"`
+	FirstNameKanji     *string                     `form:"first_name_kanji"`
+	Gender             OAuthStripeUserGender       `form:"gender"`
+	LastName           *string                     `form:"last_name"`
+	LastNameKana       *string                     `form:"last_name_kana"`
+	LastNameKanji      *string                     `form:"last_name_kanji"`
+	PhoneNumber        *string                     `form:"phone_number"`
+	PhysicalProduct    *bool                       `form:"physical_product"`
+	ProductDescription *string                     `form:"product_description"`
+	State              *string                     `form:"state"`
+	StreetAddress      *string                     `form:"street_address"`
+	URL                *string                     `form:"url"`
+	Zip                *string                     `form:"zip"`
+}
+
+// AuthorizeURLParams for creating OAuth AuthorizeURL's.
+type AuthorizeURLParams struct {
+	Params        `form:"*"`
+	AlwaysPrompt  *bool                  `form:"always_prompt"`
+	ClientID      *string                `form:"client_id"`
+	Express       *bool                  `form:"-"`
+	RedirectURI   *string                `form:"redirect_uri"`
+	ResponseType  *string                `form:"response_type"`
+	Scope         *string                `form:"scope"`
+	State         *string                `form:"state"`
+	StripeLanding *string                `form:"stripe_landing"`
+	StripeUser    *OAuthStripeUserParams `form:"stripe_user"`
+}
+
+// DeauthorizeParams for deauthorizing an account.
+type DeauthorizeParams struct {
+	Params       `form:"*"`
+	ClientID     *string `form:"client_id"`
+	StripeUserID *string `form:"stripe_user_id"`
+}
+
+// OAuthTokenParams is the set of paramaters that can be used to request
+// OAuthTokens.
+type OAuthTokenParams struct {
+	Params       `form:"*"`
+	ClientSecret *string `form:"client_secret"`
+	Code         *string `form:"code"`
+	GrantType    *string `form:"grant_type"`
+	RefreshToken *string `form:"refresh_token"`
+	Scope        *string `form:"scope"`
+}
+
+// OAuthToken is the value of the OAuthToken from OAuth flow.
+// https://stripe.com/docs/connect/oauth-reference#post-token
+type OAuthToken struct {
+	Error            OAuthError     `json:"error"`
+	ErrorDescription string         `json:"error_description"`
```

Comment:
(And just flagging this comment once more in case it was missed originally.)

Reply:
Yeah I've been trying to work through this one... The shape of the errors from OAuth are different from the rest of the API. Our error handling / wrapping is failing to unmarshal because the key "error" in the OAuth errors is the type of error and the key "error" in the rest of the API seems to be the json object with error data.

Feels like the right approach is to make the connect backend handle the error with a different error type?